pkg/activator/handler/enforce_length_handler_test.go

```diff
@@ -13,51 +13,104 @@ limitations under the License.
 package handler
 
 import (
-	"bytes"
+	"io"
+	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
+	"strings"
 	"testing"
 )
 
 func TestEnforceMaxContentLengthHandler(t *testing.T) {
 	payload := "SAMPLE PAYLOAD"
 
+	// httptest.NewRequest will set ContentLength for strings.NewReader
+	fixed := func(p string) io.Reader { return strings.NewReader(p) }
+	stream := func(p string) io.Reader { return ioutil.NopCloser(strings.NewReader(p)) }
+
 	examples := []struct {
 		label     string
 		maxUpload int
+		request   io.Reader
+		response  string
 		status    int
 	}{
 		{
-			label:     "under",
+			label:     "under with ContentLength",
 			maxUpload: len(payload) + 1,
+			request:   fixed(payload),
 			status:    http.StatusOK,
+			response:  payload,
 		},
 		{
-			label:     "equal",
+			label:     "equal with ContentLength",
 			maxUpload: len(payload),
+			request:   fixed(payload),
 			status:    http.StatusOK,
+			response:  payload,
 		},
 		{
-			label:     "over",
+			label:     "over with ContentLength",
 			maxUpload: len(payload) - 1,
+			request:   fixed(payload),
 			status:    http.StatusRequestEntityTooLarge,
+			response:  "",
+		},
+		{
+			label:     "under without ContentLength",
+			maxUpload: len(payload) + 1,
+			request:   stream(payload),
+			status:    http.StatusOK,
+			response:  payload,
+		},
+		{
+			label:     "equal without ContentLength",
+			maxUpload: len(payload),
+			request:   stream(payload),
+			status:    http.StatusOK,
+			response:  payload,
+		},
+		{
+			label:     "over without ContentLength",
+			maxUpload: len(payload) - 1,
+			request:   stream(payload),
+			status:    http.StatusOK,
+			response:  payload[0 : len(payload)-1],
 		},
 	}
 
 	for _, e := range examples {
 		t.Run(e.label, func(t *testing.T) {
+			// Return 200 and request body
 			baseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
baseHandler seems  to be test independent, and as such can be pulled out of the loop